cluster-autoscaler/core/utils.go

```diff
@@ -152,6 +154,20 @@ func GetNodeInfosForGroups(nodes []*apiv1.Node, cloudProvider cloudprovider.Clou
 	return result, nil
 }
 
+func sanitizeTemplateNodeLables(node *apiv1.Node) *apiv1.Node {
+	obj, _ := api.Scheme.DeepCopy(node)
+	newNode := obj.(*apiv1.Node)
+	newNode.Labels = make(map[string]string, len(node.Labels))
+	for k, v := range node.Labels {
+		if k != "kubernetes.io/hostname" {
+			newNode.Labels[k] = v
+		} else {
+			newNode.Labels[k] = "template-node-for-cluster-autoscaler"
```

Comment:
Shouldn't we use something unique here? Can we have 2 new nodes? For example if we do a scale up and, before new node is created, even more pods show up and we want to do another scale up?